### PR TITLE
Fixe bugs, lot of files changed

### LINKED ISSUE #19: [BUG REPORT]
**Describe the bug**
System not working 

**To Reproduce**
root | INFO | Trying to initialize the Blackfire probe
root | INFO | Blackfire not installed: passing
root | INFO | Starting queues...
root | INFO | Started queues!
root | INFO | Trying to find number of available cores
root | WARNING | Falling back to multiprocessing cpu_count to calc cores. Most likely getaffinity is not supported on your OS
root | INFO | Found 2 cores available!
root | INFO | Starting worker ID 24407
root | INFO | Started worker.
root | INFO | Starting networking worker
Traceback (most recent call last):
  File "/Users/Grant/McPy/main.py", line 238, in <module>
    main()
  File "/Users/Grant/McPy/main.py", line 221, in main
    networkingProcess.start()
  File "/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/multiprocessing/process.py", line 121, in start
    self._popen = self._Popen(self)
  File "/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/multiprocessing/context.py", line 224, in _Popen
    return _default_context.get_context().Process._Popen(process_obj)
  File "/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/multiprocessing/context.py", line 284, in _Popen
    return Popen(process_obj)
  File "/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/multiprocessing/popen_spawn_posix.py", line 32, in __init__
    super().__init__(process_obj)
  File "/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/multiprocessing/popen_fork.py", line 19, in __init__
    self._launch(process_obj)
  File "/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/multiprocessing/popen_spawn_posix.py", line 47, in _launch
    reduction.dump(process_obj, fp)
  File "/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/multiprocessing/reduction.py", line 60, in dump
    ForkingPickler(file, protocol).dump(obj)
TypeError: cannot pickle '_cffi_backend.FFI' object
root | INFO | Trying to initialize the Blackfire probe
root | INFO | Blackfire not installed: passing
root | INFO | Starting queues...
root | INFO | Started queues!
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/multiprocessing/spawn.py", line 116, in spawn_main
    exitcode = _main(fd, parent_sentinel)
  File "/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/multiprocessing/spawn.py", line 126, in _main
    self = reduction.pickle.load(from_parent)
  File "/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/multiprocessing/synchronize.py", line 110, in __setstate__
    self._semlock = _multiprocessing.SemLock._rebuild(*state)
FileNotFoundError: [Errno 2] No such file or directory

**Expected behavior**

**Screenshots**

**Host system:**
 - OS: [Macos Catalina Latest]
 - OS version: [10.15.6]
 - Python version [3.8.5/Pycharm 2020.2] 

**Additional context**

### PR DESCRIPTION
**Pull request by 0ddlyoko**
- Check if python version is above 3.8
- Import all classes with `__init__.py` files (Commented)
- Move classes/Materials.py to classes/blocks/Materials.py
- startup.py:
  - Set logging config to INFO
  - Remove --pythonlocation command line (useless)
  - Add --debug command line that set logging config to DEBUG
  - Delete mcPyTempFile.zip file after extraction
  - Convert Python 3 syntax to Python 2 to prevent an error showing on the console
- main.py:
  - Fix type (avaliCPUs => availCPUs)
  - Comment code that make errors
  - Fixes #19 
  - Fixes a lot of errors & exceptions
  - Fixes logging messages that wasn't displayed when sent from another process
  - Day time is now synchronized with the server

### CODE REVIEW

**Comment by ntoskrnl4** [comment]
> Change ip from 0.0.0.0 to 127.0.0.1 (for tests)

This will prevent anyone outside of localhost from connecting to the server, are you sure you want to do this?

The server binds to 0.0.0.0, because that address indicates that it will receive connections from anywhere. By setting it to bind to 127.0.0.1, anyone that does *not* connect using 127.0.0.1 (localhost) will not be able to connect.

Read: Undo that change, it's not what you intend to do

**Comment by 0ddlyoko** [comment]
> > Change ip from 0.0.0.0 to 127.0.0.1 (for tests)
> 
> This will prevent anyone outside of localhost from connecting to the server, are you sure you want to do this?
> 
> The server binds to 0.0.0.0, because that address indicates that it will receive connections from anywhere. By setting it to bind to 127.0.0.1, anyone that does _not_ connect using 127.0.0.1 (localhost) will not be able to connect.
> 
> Read: Undo that change, it's not what you intend to do

This is just for testing, I'll change it when this pull request is done

File: classes/__init__.py [hunk 1]
```diff
@@ -0,0 +1,10 @@
+# Import blocks
+from . import blocks
+
+# Import files
+from . import BasicClasses
+from . import Exceptions
+# Library not found, uncomment next line when this will be fixed
+#from . import PathFinder
+from . import TerrainFeatures
+from . import WorldGenerator
```
Comment (Geolykt) [inline classes/__init__.py]: same goes to here

File: .gitignore [hunk 2]
```diff
@@ -0,0 +1,5 @@
+.vscode
+__pycache__
+McPy/
+
+McPyConfig.yaml
```
Comment (Geolykt) [inline .gitignore]: you should add a newline at the end as github has some problems with it (probably can be ignored though)

Comment (0ddlyoko) [reply .gitignore]: Done, thanks for the review

File: classes/__init__.py (hunk 1 above)
Comment (0ddlyoko) [reply classes/__init__.py]: Fixed, thanks

**Comment by 0ddlyoko** [comment]
I'm ok for a review 😄

**Comment by Geolykt** [comment]
It appears to be working fine here under Windows 10 1903

**Review by Geolykt — APPROVED** [review]
Appears to be working fine, however chat is not relayed to console. I do not know if it is a consequence of the changes or if it's a mistake of the main branch

**Comment by 0ddlyoko** [comment]
> Appears to be working fine, however chat is not relayed to console. I do not know if it is a consequence of the changes or if it's a mistake of the main branch

Fixed

File: classes/__init__.py [hunk 3]
```diff
@@ -0,0 +1,10 @@
+# Import blocks
+from . import blocks
+
+# Import files
+from . import BasicClasses
```
Comment (Geolykt) [inline classes/__init__.py:5]: This structure does not seem to run.
Doing something like:
<pre>
from .BasicClasses import Chunk, Block, Region
from .WorldGenerator import WorldGenerator, GENERATORS
</pre>
Does however run. Maybe it's just on my side (although I wonder where I went wrong then), but if it's not, then it's an easy fix

**Review by Geolykt — CHANGES_REQUESTED** [review]

File: classes/__init__.py (hunk 3 above)
Comment (0ddlyoko) [reply classes/__init__.py:5]: __ init__ is used here to load files, I don't think import like your suggestion is good, we don't have to import classes but only load python files.
I'll check if files are loaded when I'll be on my computer

Comment (Geolykt) [reply classes/__init__.py:5]: Well, it's sorted now, looks as if it works, what went wrong, idk, but the issue was on my side

**Review by Geolykt — APPROVED** [review]

**Comment by 0ddlyoko** [comment]
Conflict fixed

**Comment by 0ddlyoko** [comment]
@tazz4843 could you check to this PR when you have time? I need it to be able to add more functionalities

File: startup.py [hunk 4]
```diff
@@ -33,20 +41,27 @@
 parser.add_argument("--versions",         # Command line flag to print out list of all versions
                     action="store_true",  # If this flag is defined, parsedArgs.versions will be True: else False
                     help="print out a list of all McPy versions")
-parser.add_argument("--pythonlocation",  # Command line flag to define a Python interpeter location
-                    action="store",      # Stores the result of this flag in parsedArgs.pythonlocation
-                    default=None,        # If not passed, defaults to None
-                    help="location of the Python 3.8 interpeter: defaults to the one found in the path")
+parser.add_argument("--debug",          # Command line flag to set logging in DEBUG mode
```
Comment (tazz4843) [inline startup.py:44]: Should just use the same kwarg as the --versions flag does: `action="store_true"`: it'd make the code at lines 49-54 easier to read.

File: startup.py [hunk 5]
```diff
@@ -33,20 +41,27 @@
...
+
 
 # Automagically installs all packages required
 required = {"quarry", "twisted", "cryptography", "requests"}
 installed = {pkg.key for pkg in pkg_resources.working_set}
 missing = required - installed
 if missing:
     # implement pip as a subprocess:
-    subprocess.check_call([sys.executable, '-m', 'pip', 'install', *missing])
+    # This prevents SyntaxError while launching with python2
```
Comment (tazz4843) [inline startup.py:59]: Python 2 was EOL in Jan 2020: no need to try to support it.

**Review by tazz4843 — APPROVED** [review]
Added a few comments in `startup.py`: nothing major, but just a few things to keep in mind.

File: startup.py (hunk 4 above)
Comment (0ddlyoko) [reply startup.py:44]: Done

**Review by tazz4843 — APPROVED** [review]
Much better!

**Comment by tazz4843** [comment]
Is it okay if I merge this into master?

**Comment by 0ddlyoko** [comment]
Yes, except if there is any other thing to fix or to improve ^^

**Pull request merged**